core/src/main/java/org/infinispan/interceptors/impl/BaseStateTransferInterceptor.java

```diff
@@ -237,13 +232,24 @@ private Object handleReadCommandReturn(InvocationContext rCtx, VisitableCommand
       } else if (ce instanceof AllOwnersLostException) {
          if (trace)
             getLog().tracef("All owners for command %s have been lost.", cmd);
-         // In scattered cache it might be common to lose the single owner, we need to retry. We fill find out that
+         // In scattered cache it might be common to lose the single owner, we need to retry. We will find out that
          // we can return null only after the next topology is installed. If partition handling is enabled we decide
          // only based on the availability status.
          // In other cache modes, during partition the exception is already handled in PartitionHandlingInterceptor,
          // and if the handling is not enabled, we can't but return null.
          if (cacheConfiguration.clustering().cacheMode().isScattered()) {
-            requestedTopologyId = nextTopology(cmd, currentTopologyId);
+            AllOwnersLostException aole = (AllOwnersLostException) ce;
+            // It is possible that a command arrives with higher topology id than the one of current node; such
+            // command can cause execution of another command with CACHE_MODE_LOCAL flag (not checking topology id
+            // vs. command id) and this could find no owner in the current cache topology.
```

Comment:
It was your requirement that a read command never blocks, waiting for another topology.